lib/jamaica/games/unit_game.dart

```diff
@@ -0,0 +1,258 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:maui/jamaica/state/game_utils.dart';
+import 'package:maui/jamaica/widgets/cute_button.dart';
+
+class _ChoiceDetail {
+  int choice;
+  Reaction reaction;
+  String position;
+  int currentValue;
+  bool success;
+
+  _ChoiceDetail(
+      {this.choice,
+      this.position,
+      this.reaction = Reaction.success,
+      this.currentValue = 0,
+      this.success = false});
+  @override
+  String toString() =>
+      '_ChoiceDetail(choice: $choice,  position:$position,currentValue:$currentValue,success:$success , reaction: $reaction)';
+}
+
+class UnitGame extends StatefulWidget {
+  final Map<String, int> question;
```

Comment:
Take one number as the input. You can split it internally into unit, ten, hundred